### PR TITLE
[Doc] Make doc code snippet testable [9/n] 

### PR DESCRIPTION
**Pull request by jjyao**
## Why are these changes needed?

Remove `# doctest: +SKIP`

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/util/actor_group.py [hunk 1]
```diff
@@ -96,7 +96,9 @@ def __init__(
         init_args: Optional[Tuple] = None,
         init_kwargs: Optional[Dict] = None,
     ):
-        ray._private.usage.usage_lib.record_library_usage("util.ActorGroup")
+        from ray._private.usage.usage_lib import record_library_usage
+
+        record_library_usage("util.ActorGroup")
```
Comment (bveeramani) [inline python/ray/util/actor_group.py:101]: Is this change unrelated?

File: python/ray/util/actor_pool.py [hunk 2]
```diff
@@ -33,7 +33,9 @@ def double(self, v):
     """
 
     def __init__(self, actors: list):
-        ray._private.usage.usage_lib.record_library_usage("util.ActorPool")
+        from ray._private.usage.usage_lib import record_library_usage
+
+        record_library_usage("util.ActorPool")
```
Comment (bveeramani) [inline python/ray/util/actor_pool.py:38]: Is this change unrelated?

File: python/ray/util/queue.py [hunk 3]
```diff
@@ -38,19 +38,23 @@ class Queue:
...
+                assert item == q.get()
+            # Create Queue with the underlying actor reserving 1 CPU.
+            q = Queue(actor_options={"num_cpus": 1})
     """
 
     def __init__(self, maxsize: int = 0, actor_options: Optional[Dict] = None) -> None:
-        ray._private.usage.usage_lib.record_library_usage("util.Queue")
+        from ray._private.usage.usage_lib import record_library_usage
+
+        record_library_usage("util.Queue")
```
Comment (bveeramani) [inline python/ray/util/queue.py:57]: Is this change unrelated?

File: python/ray/workflow/api.py [hunk 4]
```diff
@@ -129,24 +129,32 @@ def run(
...
+
+            flight1 = book_flight.bind("OAK", "SAN")
+            flight2 = book_flight.bind("SAN", "OAK")
+            hotel = book_hotel.bind("SAN")
+            trip = finalize_trip.bind([flight1, flight2, hotel])
+            print(workflow.run(trip))
+
+        .. testoutput::
+
+            Flight: OAK->SAN | Flight: SAN->OAK | Hotel: SAN
```
Comment (bveeramani) [inline python/ray/workflow/api.py:157]: Nice example!

File: python/ray/workflow/event_listener.py [hunk 5]
```diff
@@ -33,12 +33,15 @@ async def event_checkpointed(self, event: Event):
...
     Example Usage
     =============
-    >>> from ray import workflow
-    >>> CustomEventListener = ... # doctest: +SKIP
-    >>> event_task = workflow.wait_for_event( # doctest: +SKIP
-    ...     CustomEventListener, "topic1", "partition2")
-    >>> handle_event = ... # doctest: +SKIP
-    >>> workflow.run(handle_event.task(event_task)) # doctest: +SKIP
+    .. testcode::
+        :skipif: True
```
Comment (bveeramani) [inline python/ray/workflow/event_listener.py:37]: Would it be possible to flesh out this example? After reading the code snippets, I don't fully understand how this works.

One option would be to `testcode` the "Example definition" above and use it here.

File: python/ray/workflow/http_event_provider.py [hunk 6]
```diff
@@ -69,19 +69,21 @@ async def report_checkpointed(self, workflow_id, event, confirmation):
...
+
+        from ray.workflow.http_event_provider import HTTPEventProvider, HTTPListener
+        ray.init(address='auto', namespace='serve')
+        serve.start(detached=True)
+        event_node = workflow.wait_for_event(
+            HTTPListener, event_key='')
+        handle_event = ...
+        workflow.run_aync(handle_event.bind(event_node))
+
+        On a separate python process, it sends an event to the HTTPEventProvider.
```
Comment (bveeramani) [inline python/ray/workflow/http_event_provider.py:83]: This should be a comment or plain text, not in the code block.
```suggestion
On a separate python process, it sends an event to the HTTPEventProvider.

.. testcode::
    :skipif: True
```

File: python/ray/workflow/http_event_provider.py [hunk 7]
```diff
@@ -69,19 +69,21 @@ async def report_checkpointed(self, workflow_id, event, confirmation):
...
-    >>> handle_event = ... # doctest: +SKIP
-    >>> workflow.run_aync(handle_event.bind(event_node)) # doctest: +SKIP
-    >>>
-    >>> On a separate python process, it sends an event to the HTTPEventProvider.
-    >>> import requests
-    >>> resp = requests.post('http://127.0.0.1:8000/event/send_event/{workflow_id}',
-    ...     json={'event_key':'my_key','event_payload':'testMessage'})
-    >>>
+    .. testcode::
+        :skipif: True
```
Comment (bveeramani) [inline python/ray/workflow/http_event_provider.py:73]: Would it be possible to write this in a testable way without ellipsis?

File: python/ray/util/actor_group.py (hunk 1 above)
Comment (jjyao) [reply python/ray/util/actor_group.py:101]: It's related. It's an issue uncovered by the snippet test.

```
>>> ray._private.usage.usage_lib.record_library_usage("helo")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'ray._private' has no attribute 'usage'
```

File: python/ray/util/actor_pool.py (hunk 2 above)
Comment (jjyao) [reply python/ray/util/actor_pool.py:38]: Same as above.

File: python/ray/util/queue.py (hunk 3 above)
Comment (jjyao) [reply python/ray/util/queue.py:57]: Same as above.

**Review by fishbone — APPROVED** [review]
blindly approve.

**Review by bveeramani — APPROVED** [review]

**Pull request merged**